Tests/BackendIntegrationTests/OfflineStoreKitIntegrationTests.swift

```diff
@@ -0,0 +1,169 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  OfflineStoreKitIntegrationTests.swift
+//
+//  Created by Nacho Soto on 5/17/23.
+
+import Nimble
+@testable import RevenueCat
+import StoreKit
+import XCTest
+
+class OfflineStoreKit2IntegrationTests: OfflineStoreKit1IntegrationTests {
+
+    override class var storeKit2Setting: StoreKit2Setting { return .enabledForCompatibleDevices }
+
+}
+
+class OfflineStoreKit1IntegrationTests: BaseStoreKitIntegrationTests {
+
+    var serverIsDown: Bool = false
+    override var forceServerErrors: Bool { return self.serverIsDown }
+
+    override func setUp() async throws {
+        try await super.setUp()
+
+        await self.waitForPendingCustomerInfoRequests()
+        if #available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *) {
+            try await self.ensureEntitlementMappingIsAvailable()
+        }
+    }
+
+    func testOfferingsAreCachedInMemory() async throws {
+        let onlineOfferings = try await Purchases.shared.offerings()
+        expect(onlineOfferings.all).toNot(beEmpty())
+
+        self.serverDown()
+
+        let offlineOfferings = try await Purchases.shared.offerings()
+        expect(offlineOfferings) === onlineOfferings
+    }
+
+    func testOfferingsAreCachedOnDisk() async throws {
+        let onlineOfferings = try await Purchases.shared.offerings()
+        expect(onlineOfferings.all).toNot(beEmpty())
+
+        self.serverDown()
+        await self.resetPurchases()
+
+        let offlineOfferings = try await Purchases.shared.offerings()
+        expect(offlineOfferings.response) == onlineOfferings.response
+
+        let offering = try XCTUnwrap(offlineOfferings.current)
+        expect(offering.availablePackages.count) == onlineOfferings.current?.availablePackages.count
+        expect(offering.monthly?.storeProduct.productIdentifier) == "com.revenuecat.monthly_4.99.1_week_intro"
+    }
+
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func testOfflineCustomerInfoWithNoPurchases() async throws {
+        Purchases.shared.invalidateCustomerInfoCache()
```

Comment:
Not directly related to this PR but looking at this comment https://github.com/RevenueCat/purchases-ios/blob/main/Tests/BackendIntegrationTests/BaseBackendIntegrationTests.swift#L144-L154. Looks like that waits for get offerings to finish, but not for the get customer info request to finish I think? If so, there could possibly be race conditions of getting the result for that request after this line is executed... There might be something I'm missing though!

Reply:
Nvm! I didn't see `waitForPendingCustomerInfoRequests`